java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/FlightServerTestRule.java

```diff
@@ -231,4 +249,55 @@ public FlightServerTestRule build() {
           new RootAllocator(Long.MAX_VALUE), producer, authentication);
     }
   }
+
+  /**
+   * A middleware to handle with the cookies in the server. It is used to test if cookies are
+   * being sent properly.
+   */
+  static class MiddlewareCookie implements FlightServerMiddleware {
+
+    private final Factory factory;
+
+    public MiddlewareCookie(Factory factory) {
+      this.factory = factory;
+    }
+
+    @Override
+    public void onBeforeSendingHeaders(CallHeaders callHeaders) {
+      if (!factory.receivedCookieHeader) {
+        callHeaders.insert("Set-Cookie", "k=v");
+      }
+    }
+
+    @Override
+    public void onCallCompleted(CallStatus callStatus) {
+
+    }
+
+    @Override
+    public void onCallErrored(Throwable throwable) {
+
+    }
+
+    /**
+     * A factory for the MiddlewareCookkie.
+     */
+    static class Factory implements FlightServerMiddleware.Factory<MiddlewareCookie> {
+
+      private boolean receivedCookieHeader = false;
```

Comment:
This doesn't seem to get used.

Reply:
The `receivedCookieHeader` are used in `onBeforeSendingHeaders`. Line 266.